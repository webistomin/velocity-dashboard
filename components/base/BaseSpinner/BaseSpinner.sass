+b('base-spinner')
  display: block
  box-sizing: border-box
  border-radius: 50%
  border: rem(4px) solid rgba(var(--color-white), 0.2)
  position: relative

  &::after
    box-sizing: inherit
    width: inherit
    height: inherit
    content: ''
    +absolute-center()
    border: rem(4px) solid rgba(var(--color-white), 0.2)
    border-top: rem(4px) solid rgb(var(--color-primary))
    border-radius: inherit
    z-index: 1
    animation: spin-loading 1s linear infinite

  +m('size')
    +v('xs')
      width: rem(24px)
      height: rem(24px)

    +v('s')
      width: rem(32px)
      height: rem(32px)

    +v('m')
      width: rem(40px)
      height: rem(40px)
