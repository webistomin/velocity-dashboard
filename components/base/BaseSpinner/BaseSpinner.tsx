import Vue, { RenderContext, VNode, CreateElement, PropType } from 'vue';

import './BaseSpinner.sass';

export type IBaseSpinnerSize = 'xs' | 's' | 'm';

export interface IBaseSpinnerProps {
  size?: IBaseSpinnerSize;
}

export const BaseSpinner = Vue.extend({
  functional: true,
  props: {
    size: {
      type: String as PropType<IBaseSpinnerProps['size']>,
      default: 'xs',
    },
  },
  render(_h: CreateElement, ctx: RenderContext<IBaseSpinnerProps>): VNode {
    const { staticClass, class: cls } = ctx.data;
    const { size } = ctx.props;
    return (
      <span class={`base-spinner base-spinner_size_${size} ${staticClass || ''} ${cls || ''}`}>{ctx.children}</span>
    );
  },
});
