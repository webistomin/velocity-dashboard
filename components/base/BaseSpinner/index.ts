import { BaseSpinner } from './BaseSpinner';

export default BaseSpinner;
