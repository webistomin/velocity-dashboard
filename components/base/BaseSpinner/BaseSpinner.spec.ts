import { mount } from '@vue/test-utils';
import { BaseSpinner } from './BaseSpinner';

describe('BaseSpinner', () => {
  const factory = (propsData = {}, options = {}) => {
    return mount(BaseSpinner, {
      propsData: {
        ...propsData,
      },
      ...options,
    });
  };

  it('Matches snapshot', () => {
    const wrapper = factory().html();
    expect(wrapper).toMatchSnapshot();
  });

  it('Set class from size prop', () => {
    const expectedResult = ['base-spinner_size_m'];
    const wrapper = factory({
      size: 'm',
    });
    const classes = wrapper.classes();
    expect(classes).toEqual(expect.arrayContaining(expectedResult));
  });

  it('Set default size class if size prop is not provided', () => {
    const expectedResult = ['base-spinner_size_xs'];
    const wrapper = factory();
    const classes = wrapper.classes();
    expect(classes).toEqual(expect.arrayContaining(expectedResult));
  });
});
