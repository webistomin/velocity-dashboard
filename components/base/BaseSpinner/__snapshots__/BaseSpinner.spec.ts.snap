// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseSpinner Matches snapshot 1`] = `"<span class=\\"base-spinner base-spinner_size_xs  \\"></span>"`;
